### PR TITLE
Int64common

### PR DESCRIPTION
**Pull request by xanlpz**
Per previous reviews:

- Just use the stdint.h int64 min value everywhere.
- Fallback to the !_mul128 implementation on !windows, this seems simpler for now.

### CODE REVIEW

File: lib/common/common/Int64Math.cpp [hunk 1]
```diff
@@ -3,12 +3,15 @@
 // Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 //-------------------------------------------------------------------------------------------------------
 #include "CommonCommonPch.h"
-#include "Common\Int64Math.h"
-#include <intrin.h>
+#include "common/Int64Math.h"
+#include <stdint.h>
 
+#if _WIN32 || _WIN64
```
Comment (jianchun) [inline lib/common/common/Int64Math.cpp]: ```#ifdef _WIN32``` suffices ([see msdn](https://msdn.microsoft.com/en-us/library/b0084kay.aspx))

File: lib/common/common/Int64Math.cpp [hunk 2]
```diff
@@ -21,7 +24,7 @@ Int64Math::Add(int64 left, int64 right, int64 *pResult)
 bool
 Int64Math::Mul(int64 left, int64 right, int64 *pResult)
 {
-#if _M_X64
+#if _M_X64 && !PLATFORM_UNIX
```
Comment (jianchun) [inline lib/common/common/Int64Math.cpp]: As discussed in #192, using ```defined(_MSC_VER)``` is probably preferred. ```PLATFORM_UNIX``` is introduced for ```PAL``` and we haven't used it in ChakraCore sources yet.

**Comment by jianchun** [comment]
> I thought the focus was first to get things compiling and then improve from there

I agree. I tried to search it in code and can't find usage in interpreter mode. This is probably only used by code generator to evaluate constants and not affecting runtime perf.

**Comment by obastemur** [comment]
> Whether we want to just inline assembly...

Prepared a unix/linux version of `__mul128` https://gist.github.com/obastemur/48125aacb960a5b66730

@jianchun just saw that. So can we put a TODO notice there ? Even if the focus is making the code compile first better not to miss what needs to be done later.

**Comment by xanlpz** [comment]
OK, I changed things to check for _MSC_VER, only check for _WIN32, and added a TODO for mul128 in !windows.

**Comment by jianchun** [comment]
@obastemur  Thanks for creating inline assembly version! I've opened an issue to track it.
@xanlpz  Thank you for this PR and updates!

**Pull request merged**